### ISSUE TITLE
Internal element state overrides updating props

### ISSUE THREAD

**Issue by EliRibble** [DominicTobias/react-image-crop#24]
My use-case may not be normally covered by react-image-crop, but it seems like a normal workflow. I want to create a crop widget and occasionally update the crop parameters based on saved data on a server. This means I need to set the `crop` property once, then some time later set it to something else. It appears that the internal `crop` state is overriding the property at:

```
 36   getInitialState(props = this.props) {
 37     const crop = assign({}, this.defaultCrop, props.crop, this.state ? this.state.crop : {});
 38
 39     this.cropInvalid = (!crop.width || !crop.height);
 40
 41     return {
 42       crop: crop
 43     };
 44   },
```

This means that after initially setting `crop` via props it can't be set again making my use-case not work.

I've created an updated version of the plugin that does not keep `crop` in internal state. Everything I tested seems to work, but I didn't read all of the code to understand what, if any, feature I might have ruined. I wanted to open this ticket to start a conversation on the topic to see if a PR would be useful or not.

**Comment by DPr00f** [comment]
I really don't get why you assume that the internal state prevents the workflow you mentioned.
Please check the method `componentWillReceiveProps` you can tell at that point that the crop state will be updated with the new values.
I agree it might not be the best solution, but it works.

**Comment by EliRibble** [comment]
I’m not assuming - that was my experience when using the component in my application. When I have time I may throw together a minimal app that reproduces the problem.

**Comment by DPr00f** [comment]
Ah, I didn't realise that was your experience. I thought you you just
wanted to discuss it. How are you updating it now then? You changed the
crop state to be a prop?

**Comment by DominicTobias** [comment]
It's supposed to work - i.e. if the parent component re-renders and the crop prop is different then it should update. Feel free to paste the parts of relevant code.

**Comment by bhough** [comment]
I've also had the issue. The problem resides in the `getInitialState` method. The newProps are spread across the existing state using Object.Assign. The order that the properties are assigned causes new props to be overwritten by existing ones if they are already present in the state. 

I created a PR that rearranges these so that the new props are always assigned last, thus overriding existing values in the state.

https://github.com/DominicTobias/react-image-crop/pull/25

**Comment by DominicTobias** [comment]
Thanks for spotting the issue @bhough, I'll take a look now

**Comment by DominicTobias** [comment]
Released under 0.1.5
